cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptLookup.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.etl.transform;
+
+import co.cask.cdap.etl.api.Lookup;
+import co.cask.cdap.etl.api.LookupTableConfig;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * {@link Lookup} implementation for {@link ScriptTransform} that processes the {@link LookupTableConfig}.
+ */
+public class ScriptLookup implements Lookup<Object> {
```

Comment:
So this class is only used by `ScriptLookupProvider` right? Does it intended to be a public class? If not, better make it a private static inner class. It's a bit confusion to have `ScriptLookup` and `ScriptLookupTable` as top level public class.

Actually why need ScriptLookup and ScriptLookupTable? Shouldn't they be one?